x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -0,0 +1,460 @@
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobId } = request.params;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.openJob', {
+          jobId,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/:jobId/_close Close specified job
+   * @apiName CloseAnomalyDetectorsJob
+   * @apiDescription Closes an anomaly detection job.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_close',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const options: { jobId: string; force?: boolean } = {
+          jobId: request.params.jobId,
+        };
+        const force = request.query.force;
+        if (force !== undefined) {
+          options.force = force;
+        }
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.closeJob', options);
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {delete} /api/ml/anomaly_detectors/:jobId Delete specified job
+   * @apiName DeleteAnomalyDetectorsJob
+   * @apiDescription Deletes specified anomaly detection job.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.delete(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const options: { jobId: string; force?: boolean } = {
+          jobId: request.params.jobId,
+        };
+        const force = request.query.force;
+        if (force !== undefined) {
+          options.force = force;
+        }
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.deleteJob', options);
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/_validate/detector Validate detector
+   * @apiName ValidateAnomalyDetector
+   * @apiDescription Validates specified detector.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/_validate/detector',
+      validate: {
+        body: schema.any(),
```

Comment:
seems like body schema validation is missing 🙂

Reply:
I don't see this one being used anywhere 🤔  so I held off adding a schema as it would make it pretty restrictive.